src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -34,7 +34,9 @@
             throw new DataProcessingException("Failed to get cart of user with Id = "
                     + userId, e);
         }
-        shoppingCart.setProducts(getProductsFromCart(shoppingCart.getId()));
+        if (shoppingCart.getProducts() != null) {
```

Comment:
can getProducts() NOT return `null` at this point?
reconsider your null check please

Reply:
yes, done, not sure if correct